zerver/actions/streams.py

```diff
@@ -893,6 +894,64 @@ def send_subscription_remove_events(
     )
 
 
+def send_user_remove_events_to_inaccessible_users(
+    realm: Realm, altered_user_dict: Dict[UserProfile, Set[int]]
+) -> None:
+    altered_stream_ids: Set[int] = set()
+    altered_users = list(altered_user_dict.keys())
+    for stream_ids in altered_user_dict.values():
+        altered_stream_ids |= stream_ids
+
+    users_involved_in_dms = get_users_involved_in_dms_with_target_users(altered_users, realm)
+    subscribers_of_altered_user_subscriptions = get_subscribers_of_target_user_subscriptions(
+        altered_users
+    )
+
+    non_guest_user_ids = active_non_guest_user_ids(realm.id)
+
+    subscribers_dict = get_user_ids_for_streams(altered_stream_ids)
+
+    for user in altered_users:
+        users_in_unsubscribed_streams: Set[int] = set()
+        for stream_id in altered_user_dict[user]:
+            users_in_unsubscribed_streams = (
+                users_in_unsubscribed_streams | subscribers_dict[stream_id]
+            )
+
+        users_who_can_access_altered_user = (
+            set(non_guest_user_ids)
+            | subscribers_of_altered_user_subscriptions[user.id]
+            | users_involved_in_dms[user.id]
+        )
+
+        subscribers_without_access_to_altered_user = (
+            users_in_unsubscribed_streams - users_who_can_access_altered_user
+        )
+
+        if subscribers_without_access_to_altered_user:
+            event_remove_user = dict(
+                type="realm_user",
+                op="remove",
+                person=dict(user_id=user.id, full_name=str(UserProfile.INACCESSIBLE_USER_NAME)),
+            )
+            send_event_on_commit(
+                realm, event_remove_user, list(subscribers_without_access_to_altered_user)
+            )
+
+        if user.is_guest:
+            users_inaccessible_to_altered_user = users_in_unsubscribed_streams - (
+                subscribers_of_altered_user_subscriptions[user.id] | users_involved_in_dms[user.id]
+            )
+
+            for user_id in users_inaccessible_to_altered_user:
+                event_remove_user = dict(
+                    type="realm_user",
+                    op="remove",
+                    person=dict(user_id=user_id, full_name=str(UserProfile.INACCESSIBLE_USER_NAME)),
```

Comment:
I think we can remove the `full_name` field here since I guess existing clients do not use this event (need to recheck though). Have kept it for now as we used to send it in the deprecated event.

Reply:
Yeah, I expect we can remove that. Please post in "api design" about this to confirm that theory.